### PR TITLE
Adding csrf package, csrf token stored in cookie, post method ensures…

### PR DESCRIPTION
**Pull request by shaunganley**
… that token in view is the same as cookie when submitting

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -56,6 +56,7 @@ Clone or copy the source into your own project. There are sections marked with `
   * headers - added [helmet](https://www.npmjs.com/package/helmet)
   * CQRS - TODO
```
Comment (stevenalexander) [inline README.md]: remove

File: README.md [hunk 2]
```diff
@@ -56,6 +56,7 @@ Clone or copy the source into your own project. There are sections marked with `
...
   * Request sanitisation - TODO
+  * cross-site scripting - added [csurf](https://github.com/expressjs/csurf) package which uses cookies to protect against [XSS](https://www.owasp.org/index.php/Cross-site_Scripting_(XSS)), see [here](https://github.com/pillarjs/understanding-csrf) for how this works
```
Comment (stevenalexander) [inline README.md]: Can you increase the documentation so it's more detailed? I'm thinking for people who are just looking at this repo fresh and want to understand what this is for and why. I broke out the Configuration readme section in a new paragraph like [this](https://github.com/KainosSoftwareLtd/govuk_nodejs_typescript_example/pull/1/commits/2f33f4a47ec49a70b0c273d2b6f10ae259c5b8d3). Details like what you need to add to each form view to support this and how you handle whitelisting requests.

File: src/app.ts [hunk 3]
```diff
@@ -40,6 +42,8 @@ app.use(function (req, res, next) {
   res.locals.asset_path = '/public/'
   res.locals.serviceName = 'Example service' // TODO replace with service name
   res.locals.releaseVersion = 'v0.1' // TODO replace with package.json version
+  res.locals.csrfToken = req.csrfToken()
```
Comment (stevenalexander) [inline src/app.ts]: is this actually validating the token used in posts? You can test by using chrome dev tools to either clear the CSRF field or copy and edit the form post request as curl and repeat (simulating a repeated request).  See [here](https://github.com/ministryofjustice/apvs-external-web/blob/develop/app/app.js#L155) for how this was added to APVS.

Comment (shaunganley) [reply src/app.ts]: yep, included details in readme if you want to test this out yourself locally

File: README.md (hunk 1 above)
Comment (shaunganley) [reply README.md]: done, this is removed as csrf covers this off

**Pull request merged**